sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/LegacySpanProcessorMetrics.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.export;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.MeterProvider;
+import javax.annotation.Nullable;
+
+/** Span processor metrics defined before they were standardized in semconv. */
+final class LegacySpanProcessorMetrics implements SpanProcessorMetrics {
+  private static final AttributeKey<String> SPAN_PROCESSOR_TYPE_LABEL =
+      AttributeKey.stringKey("processorType");
+  private static final AttributeKey<Boolean> SPAN_PROCESSOR_DROPPED_LABEL =
+      AttributeKey.booleanKey("dropped");
+  // Legacy metrics are only created for batch span processor.
+  private static final String SPAN_PROCESSOR_TYPE_VALUE = BatchSpanProcessor.class.getSimpleName();
+
+  private final Meter meter;
+  private final Attributes standardAttrs;
+  private final Attributes droppedAttrs;
+
+  private final LongCounter processedSpans;
+
+  LegacySpanProcessorMetrics(MeterProvider meterProvider) {
+    meter = meterProvider.get("io.opentelemetry.sdk.trace");
+
+    processedSpans =
+        meter
+            .counterBuilder("processedSpans")
+            .setUnit("1")
+            .setDescription(
+                "The number of spans processed by the BatchSpanProcessor. "
+                    + "[dropped=true if they were dropped due to high throughput]")
+            .build();
+
+    standardAttrs =
+        Attributes.of(
+            SPAN_PROCESSOR_TYPE_LABEL,
+            SPAN_PROCESSOR_TYPE_VALUE,
+            SPAN_PROCESSOR_DROPPED_LABEL,
+            false);
+    droppedAttrs =
+        Attributes.of(
+            SPAN_PROCESSOR_TYPE_LABEL,
+            SPAN_PROCESSOR_TYPE_VALUE,
+            SPAN_PROCESSOR_DROPPED_LABEL,
+            true);
+  }
+
+  /** Records metrics for spans dropped because a queue is full. */
+  @Override
```

Comment:
#nit: not need for javadoc which repeats the same from the parent in SpanProcessorMetrics